site/app/controllers/grading/ElectronicGraderController.php

```diff
@@ -63,11 +63,96 @@ public function RandomizePeers($gradeable_id) {
         5 - Shift the random order by the offsets to create the matrix, with no duplicates, and exactly Y assignments and & graders for each student.  no student grades self.
         */
         $number_to_grade = $_POST['number_to_grade'];
+        $restrict_to_registration = $_POST['restrict_to_registration'];
         $gradeable = $this->tryGetGradeable($gradeable_id);
         if ($gradeable === false) {
             $this->core->addErrorMessage('Invalid Gradeable!');
             $this->core->redirect($this->core->buildCourseUrl());
         }
+        /* If Restrict to Registration checkbox is checked, then the randomised peer assignments should be restricted to each registration section" */
+        if ($restrict_to_registration == "checked") {
+            $sections = $this->core->getQueries()->getRegistrationSections();
+            foreach ($sections as $i => $section) {
+                $sections[$i] = $section['sections_registration_id'];
+            }
+            /* Duplicating the entire functionality of randomization, but for each registration section rather than whole class" */
+            $final_grading_info = [];
+            foreach ($sections as $section) {
```

Comment:
What is the purpose of the above for loop? It seems like both are just iterating through the associative array
Is it possible to do something like:
```php
foreach( $sections as $key => $value ){
 $s = $value['sections_registartion_id']
```
Could you do something like ^
To avoid the additional for loop

Reply:
it works simply to get students from a registration section (Individual):
$students = $this->core->getQueries()->getUsersByRegistrationSections([$section]);